src/prefect/automations.py

```diff
@@ -0,0 +1,114 @@
+from typing import Optional
+from uuid import UUID
+
+from pydantic import Field
+from typing_extensions import Self
+
+from prefect.client.utilities import get_or_create_client
+from prefect.events.schemas.automations import AutomationCore
+from prefect.utilities.asyncutils import sync_compatible
+
+
+class Automation(AutomationCore):
+    id: Optional[UUID] = Field(default=None, description="The ID of this automation")
```

Comment:
:fire: I think this is the best way to approach it to keep the original `Automation` schema tight.